lighthouse-cli/bin.ts

```diff
@@ -249,6 +249,39 @@ function handleError(err: LighthouseError) {
   }
 }
 
+function saveResults(results: Results,
+                     flags: {output: any, outputPath: string, saveArtifacts: boolean, saveAssets: boolean}) {
+    let promise = Promise.resolve(results);
+    const cwd = process.cwd();
+    // Use the output path as the prefix for all generated files.
+    // If no output path is set, generate a file prefix using the URL and date.
+    const configuredPath = !flags.outputPath || flags.outputPath === 'stdout' ?
+        assetSaver.getFilenamePrefix(results) :
+        flags.outputPath.replace(/\.\w{2,4}$/, '');
+    const resolvedPath = path.resolve(cwd, configuredPath);
+    const pathWithBasename = resolvedPath.includes(cwd) ?
+        path.relative(cwd, resolvedPath) : resolvedPath;
+
+    const audits = results.audits;
+    const artifacts = results.artifacts;
+    // remove artifacts from result so reports won't include artifacts.
+    results = Object.assign({}, results, {artifacts: undefined});
```

Comment:
only changes in the sense that the original results object is preserved, if we never intend artifacts on being consumed along with the rest of the results I'd advocate splitting them into distinct sets in the lighthouse return object. Since that'd probably break others at this point, I'll restructure a bit more instead.